### PR TITLE
Added tests for Jekyll:StaticFile

### PR DESCRIPTION
**Pull request by mdenhoedt**
Added tests for `Jekyll:StaticFile` this should solve issue [#3626](https://github.com/jekyll/jekyll/issues/3626).

### CODE REVIEW

**Comment by envygeeks** [comment]
Are you sure this is not covered by a higher level test in features/ ?

**Comment by mdenhoedt** [comment]
The test coverage for `StaticFile` is already nearly 100%. However, I think it is a good idea to have unit tests for it too, but it's up to you :smile:.

**Comment by envygeeks** [comment]
I think it's a good idea to cover it.  Anybody else have any objects? /cc @jekyll/core

File: test/test_static_file.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
+require 'helper'
+
+class TestStaticFile < JekyllUnitTest
+    def make_dummy_file(file_name)
+        temp_file = File.new(file_name, "w")
+        temp_file.puts("some content")
+        temp_file.close
```
Comment (parkr) [inline test/test_static_file.rb]: Why not:

```ruby
File.open(filename, "w") do |f|
  f.puts "some content"
end
```

?

File: test/test_static_file.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    end
+
+    def setup_static_file(base, dir, name)
+        @static_file = StaticFile.new(@site, base, dir, name)
+    end
+
+    context "A StaticFile" do
+        setup do
+            clear_dest
+            @site = Site.new(site_configuration)
```
Comment (parkr) [inline test/test_static_file.rb]: I believe you can use `fixture_site` here!

**Comment by parkr** [comment]
I also think it's a good idea to cover it. :)

**Comment by parkr** [comment]
There's definitely an opportunity to DRY up this file (a lot of repetition right now). Would you mind tackling that?

**Comment by mdenhoedt** [comment]
Thanks for the feedback. I am not an experienced Ruby programmer and I will definitely try to improve my pull request.

File: test/test_static_file.rb (hunk 1 above)
Comment (envygeeks) [inline test/test_static_file.rb]:
```ruby
File.write(filename, "some content")
```

Comment (parkr) [inline test/test_static_file.rb]: ^^ that also works :)

**Comment by parkr** [comment]
No problem! Let us know if you have any questions. :)

**Comment by envygeeks** [comment]
:heart:

**Pull request merged**